code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -186,6 +184,30 @@ Made by Xhuis
 						  under their command...</span>")
 	return 1
 
+/datum/game_mode/proc/remove_shadowling(datum/mind/ling_mind)
+	if(!istype(ling_mind) || !(ling_mind in shadows)) return 0
+	update_shadow_icons_removed(ling_mind)
+	shadows.Remove(ling_mind)
+	ling_mind.current.attack_log += "\[[time_stamp()]\] <span class='danger'>Deshadowlinged</span>"
+	ling_mind.special_role = null
+	for(var/obj/effect/proc_holder/spell/S in ling_mind.spell_list)
+		ling_mind.remove_spell(S)
```

Comment:
Yes. I believe that silicons have their spells removed regardless - the spell removal is to remove a shadowling's abilities in the case that a carbon is somehow de-shadowed.